src/ventus/wm/windowmanager.js

```diff
@@ -102,9 +108,21 @@ function($, Window, View, DefaultMode, ExposeMode, FullscreenMode) {
 		get overlay() {
 			return this._overlay;
 		},
+		
+		set margin(marginObj) {
+			if (typeof marginObj !== 'object') {
+				return;
+			}
+			this._margin = marginObj;
```

Comment:
This will need to be changed and instead merge the new values in to the current object in order for the window's reference to be up to date at all times.